modin/backends/pandas/query_compiler.py

```diff
@@ -2569,17 +2609,89 @@ def compute_groupby(df):
                 # issues with extracting the index.
                 except (DataError, TypeError):
                     result = pandas.DataFrame(index=grouped_df.size().index)
+                if isinstance(result, pandas.Series):
+                    result = result.to_frame("__reduced__")
```

Comment:
Whether do we always need to set col_name=__reduced__?

Reply:
No, if `result.name is not None` `col_name` shouldn't be `"__reduced__"`, fixed it, thanks